File: Client/hms/src/app/models/FilterObject.ts
import { User } from './user.model';

export class FilterObject {
  limit: number | undefined;
  page: number = 1; 
  firstName: string = ''; 
  lastName: string = ''; 
  specialization: string = ''; 
  email: string = '';
  patientId: number | null = null; 
  doctorId: number | null = null; 
  status: string = ''; 
  doctorFlag: boolean = false; 
  patientFlag: boolean = false; 
  appointmentDate: number | null = null;
  appointmentType:string ='';
  id:number| undefined 
}

File: Server/hms/src/main/java/com/hms/common/dto/FilterObject.java
package com.hms.common.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
public class FilterObject {
    private int limit;
    private int page;

    private String firstName;
    private String lastName;
    private String specialization;
    private String email;
    private Integer patientId;
    private Integer doctorId;
    private String status;

    private boolean doctorFlag;
    private boolean patientFlag;

    private String itemName;
    private String itemCode;
    private Date appointmentDate;
    private String appointmentType;
    private Integer id;

}
